lib/Runtime/Library/JSONStack.cpp

```diff
@@ -37,22 +54,68 @@ namespace JSON
     {
         if (bJsObject)
         {
-            return jsObjectStack.Push(data);
+            bool result = true;
```

Comment:
Consider refactoring this into an inline function- something like `PushObject(data, jsObjectStack, &STACK_VALUE)/PushObject(data, domObjectStack, &DOM_VALUE)`